lib/meals.js

```diff
@@ -1,26 +1,111 @@
 module.exports = {totalCalories}
 
-
 var api_url = "https://obscure-harbor-85447.herokuapp.com/api/v1"
 
 $(document).ready(function(){
+  renderDiary()
+  renderFood()
+  removeFood()
+  addFoodToMeal()
+})
+
+function addFoodToMeal(){
+  $("#add-selected").on("click", function(event){
+    var foodId = event.target.getAttribute("foodId")
+    var mealId = event.target.getAttribute("mealid")
+    var checkedFoods = $("input:checked")
+    $.each(checkedFoods, function(index, food){
+      $.ajax({
+      type: "POST",
+      url: api_url + "/meals/" + mealId + "/foods/" + food.getAttribute("foodId")
+      })
+      .then(function(){
+        renderDiary()
+        renderFood()
+
+      })
+      .catch(function(error){
+        console.error(error)
+      })
+      console.log(food)
+    })
+  })
+}
+
+
+
+function renderFood(){
+  $.ajax({
+  type: "GET",
+  url: api_url + "/foods"
+  })
+  .then(function(foods){
+    $('#foods-table').children().html('');
+    renderFoodTable(foods)
+  })
+  .catch(function(error){
+    console.error(error)
+  })
+}
+
+
+function renderFoodTable(foods){
+  $("#foods-table-headers").append("<h3 class='text-center'>Foods</h3><br>"
+                            + "<p>Add Selected to:</p>")
+
+  $("#foods-table").append("<table class='table table-bordered'>"
+                          + "<thead><tr><th></th><th>Name</th><th>Calories</th></tr></thead><tbody>"
+                          + generateFoodRows(foods)
+                          + "</tbody></table>")
+}
+
+function removeFood(){
+  $("#meals").on("click", function(event){
+    var foodId = event.target.getAttribute("foodId")
+    var mealId = event.target.getAttribute("mealId")
+    $.ajax({
+    type: "DELETE",
+    url: api_url + "/meals/" + mealId + "/foods/" + foodId
+    })
+    .then(function(){
+
+      renderDiary()
+    })
+    .catch(function(error){
+      console.error(error)
+    })
+  })
+}
+
+function renderDiary(){
   $.ajax({
   type: "GET",
   url: api_url + "/meals"
   })
   .then(function(meals){
-    renderMeals(meals);
-    renderTotals(meals);
-
+    $('#meals').children().html('');
+    $('#totals-table').children().html('');
+    $('#add-selected').children().remove();
```

Comment:
since these three lines of clearing the html don't necessarily depend on the function that we're within, could we extract them out into another method and just refer to it here?

Reply:
For sure, that was on my refactor list but I'll just make that change now.